### DISCUSSION TITLE
POETRY_VIRTUALENVS_IN_PROJECT being ignored or am I doing something wrong?

### DISCUSSION THREAD

**Discussion by mkvlrn-cm42** [python-poetry/poetry#9407] [Q&A]
According to [docs](https://python-poetry.org/docs/configuration#virtualenvsin-project), setting the `POETRY_VIRTUALENVS_IN_PROJECT` environment variable to a true value forces poetry to create `.venv` directory inside the project.

I have set that variable to `1` and `true` and it was ignored both times when creating a new project (`poetry new my-project`) and then activating the shell (`poetry shell` inside the created directory), it just creates the virtualenv in the cache directory: `~/.cache/pypoetry/virtualenvs/my-project-HASH-py3.12`.

It works fine when I create a `poetry.toml` file with the correct setting there.

What could be the problem here?

Python: 3.12.3
Poetry: 1.8.3
env being set in my `.zshrc` file and present/valid when running poetry commands

**Comment by dimbleby** [comment]
it works fine, you must be doing something wrong

```console
~/foo$ poetry config virtualenvs.in-project
null

~/foo$ POETRY_VIRTUALENVS_IN_PROJECT=true poetry install
Creating virtualenv foo in /home/dch/foo/.venv
...
```

**Reply by mkvlrn-cm42** [reply]
when setting the value the way you did it works:

![image](https://github.com/python-poetry/poetry/assets/6718750/ecae33ea-e451-4492-8a5d-1edad0521da7)

but relying on the variable set by .zshrc still won't do it properly, even though it is set correctly:

![image](https://github.com/python-poetry/poetry/assets/6718750/1a32b6bc-8fc7-483a-9f11-1b31834b98d0)

I guess I'll just create an alias to set the var when using poetry, but I was wondering if there's something wonky with using env vars set elsewhere

**Reply by rafalkrupinski** [reply]
try
```sh
export POETRY_VIRTUALENVS_IN_PROJECT=true
```

**Comment by clintonroy** [comment]
How are you setting it? Please always show code. My guess is you're setting it (for the current process, the shell) but not exporting it (for the subprocess, poetry).

**Reply by kimbasn** [reply]
I faced the same behavior as @mkvlrn. Even when I exported the variable, Poetry ignored it and created the virtual environment in the cache directory.

I finally set config variable virtualenvs.in to true and it works as expected. 
```bash
poetry config virtualenvs.in-project true
```
Any new project has its virtual environment inside the project directory.
